internal/executors/http/oauth_header_generator.go

```diff
@@ -172,8 +167,8 @@ func urlEncoded(query string) string {
 // TODO: TOTP should be considered as part of caching key here as well
 func generateCachingKey(tokenUrl string, clientId string, clientSecret string, requestBody string) string {
 	h := sha256.New()
-	v := fmt.Sprintf(CACHING_KEY_FORMAT, tokenUrl, clientId, clientSecret, requestBody)
+	str := fmt.Sprintf(CACHING_KEY_FORMAT, tokenUrl, clientId, clientSecret, requestBody)
 
-	h.Write([]byte(v))
+	h.Write([]byte(str))
```

Comment:
You can condense this into:
`h.Write(fmt.Appendf(nil, CACHING_KEY_FORMAT, tokenUrl, clientId, clientSecret, requestBody))`

And avoid creating a string to then cast to []byte.

Reply:
Neat! I did not know about the `Appendf` function